force-app/main/default/classes/PlannedSalesTriggerHelper.cls

```diff
@@ -0,0 +1,138 @@
+                        }                             
+                    }
+                } 
+            }
+ 
+        }
+        if(!plannedSalesToUpdate.isEmpty()){
+            update plannedSalesToUpdate.values();
+        }
+
+    }
+    public static void updatePlannedSales(List<Planned_Sales__c> newPlannedSales,Map<Id,Planned_Sales__c> oldPlannedSalesMap){
+        
+        //Map<Integer,Integer> monthAndYear=new Map<Integer,Integer>();
+        Set<Integer> month=new Set<Integer>();
+        Set<Integer> year=new Set<Integer>();
+        Map<String,Planned_Sales__c> plannedSales=new Map<String,Planned_Sales__c>();
+        
+        
+        Map<Id,Planned_Sales__c> plannedSalesToUpdate=new Map<Id,Planned_Sales__c>();
+        
+        if(!newPlannedSales.isEmpty()){
+            for(Planned_Sales__c newPlannedSale:newPlannedSales){
+                Planned_Sales__c oldPlannedSale=oldPlannedSalesMap.get(newPlannedSale.Id);
+                
+                if(newPlannedSale.Year__c!=oldPlannedSale.Year__c || newPlannedSale.Month__c!=oldPlannedSale.Month__c 
+               
+                   || newPlannedSale.Car_Type__c!=oldPlannedSale.Car_Type__c){
+                       
+                       String key=newPlannedSale.Month__c+'_'+newPlannedSale.Year__c+'_'+newPlannedSale.Car_Type__c;
+                       //monthAndYear.put((Integer) newPlannedSale.Month__c,(Integer) newPlannedSale.Year__c);
+                       newPlannedSale.Count_of_Cars_Sold__c=0.00;
+                       newPlannedSale.Total_Amount_Earned__c=0.00;
+                       month.add((Integer) newPlannedSale.Month__c);
+                       year.add((Integer) newPlannedSale.Year__c);
+                       plannedSales.put(key,newPlannedSale);
+                   }
+            }
+            if(month!=null && year!=null){
+            for(Opportunity opportunity:[SELECT Id,CloseDate,StageName,Amount,
+                                         (SELECT Id,Product2.Condition__c FROM OpportunityLineItems) 
+                                         FROM Opportunity 
+                                         WHERE CALENDAR_MONTH(CloseDate) IN:month
+                                         AND StageName='Closed Won'
+                                         AND CALENDAR_YEAR(CloseDate) IN:year]){
+                 
+                 String key=opportunity.CloseDate.month()+'_'+opportunity.CloseDate.year();                           
+                 for(OpportunityLineItem oli:opportunity.OpportunityLineItems){
+                    String key2=key+'_'+oli.Product2.Condition__c;
+                     
+                     
+                     if(plannedSales.containsKey(key2)){
+                         
+                         Planned_Sales__c plannedSale=plannedSales.get(key2);
+                         if(plannedSale!=null){
+                             plannedSale.Total_Amount_Earned__c+=opportunity.Amount;
```

Comment:
Adding Opportunity line item amount on each LineItem iteration will not calculate if correctly

If you have 10 cars each 100k -> Opp.Amount = 1 mil, 10 iterations will add 10 mil instead of 1 min

Another thing with Count_of_Cars_Sold__c. You have to only add cars with same Condition as Car Type on planned sales